scout/server/blueprints/cases/controllers.py

```diff
@@ -566,131 +562,63 @@ def check_outdated_gene_panel(panel_obj, latest_panel):
-    """
-    evaluated_variants = {vt: [] for vt in CASE_REPORT_VARIANT_TYPES}
-    # We collect all causatives (including the partial ones) and suspected variants
-    # These are handeled in separate since they are on case level
-    for var_type in ["causatives", "suspects", "partial_causatives"]:
-        # These include references to variants
-        vt = "_".join([var_type, "detailed"])
-        for var_id in case_obj.get(var_type, []):
-            variant_obj = store.variant(var_id)
-            if not variant_obj:
+    def _get_decorated_var(var_obj: dict) -> dict:
+        return variant_decorator(
+            store=store,
+            institute_id=institute_obj["_id"],
+            case_name=case_obj["display_name"],
+            variant_id=None,
+            variant_obj=var_obj,
+            add_other=False,
+            get_overlapping=False,
+            variant_type=var_obj["category"],
+            institute_obj=institute_obj,
+            case_obj=case_obj,
+        )["variant"]
+
+    evaluated_variants_by_type: Dict[str, list] = {vt: [] for vt in CASE_REPORT_VARIANT_TYPES}
+
+    # Collect causative, partial causative and suspected variants
+    for eval_category, case_key in CASE_REPORT_VARIANT_TYPES.items():
+        for var_id in case_obj.get(case_key, []):
+            var_obj = store.variant(document_id=var_id)
+            if not var_obj:
                 continue
-            if var_type == "partial_causatives":  # Collect associated phenotypes
-                variant_obj["phenotypes"] = [
-                    value for key, value in case_obj["partial_causatives"].items() if key == var_id
-                ][0]
-            evaluated_variants[vt].append(variant_obj)
+            if case_key == "partial_causatives":
+                var_obj["phenotypes"] = case_obj["partial_causatives"][var_id]
+            evaluated_variants_by_type[eval_category].append(_get_decorated_var(var_obj=var_obj))
 
-    ## get variants for this case that are either classified, commented, tagged or dismissed.
+    # Collect all evaluated variants except causative, partial causative and suspected variants
     for var_obj in store.evaluated_variants(
         case_id=case_obj["_id"], institute_id=institute_obj["_id"]
     ):
-        # Check which category it belongs to
-        for vt in CASE_REPORT_VARIANT_TYPES:
-            keyword = CASE_REPORT_VARIANT_TYPES[vt]
-            # When found we add it to the category
-            # Each variant can belong to multiple categories
-            if keyword not in var_obj:
-                continue
-            evaluated_variants[vt].append(var_obj)
-
-    data["variants"] = {}
-
-    for var_type in evaluated_variants:
-        decorated_variants = []
-        for var_obj in evaluated_variants[var_type]:
-            # We decorate the variant with some extra information
-            filtered_var_obj = {}
-            for feat in VARIANT_REPORT_VARIANT_FEATURES:
```

Comment:
@dnil this line made sure that the variant had the keys specified in VARIANT_REPORT_VARIANT_FEATURES (acmg_classification among them). The key was there but the value was None

Reply:
And then the template test would work. It checked if there was a defined `acmg_classification.short_whatever`, and that wouldn't be defined for a `None` key, but not illegal enough to crash. Which not having `acmg_classification` when reaching down into its keys would be. Ok, then we have it!?

Reply:
I dont think we need to keep the empty keys, do we, as long as we are careful to check them?

Reply:
Yes! Perhaps let's make sure that using any of the other case or variant keys won't have the same effect. We can do a quick check tomorrow, but al least now we have a list of things to check.

Reply:
> I dont think we need to keep the empty keys, do we, as long as we are careful to check them?

Nono I would check that they exist instead

Reply:
I think we should be fine with all he other keys. I gave them a quick check anyway, but I'm sure you remember that the report in the beginning used to work with case and variants objects fished from the database and passed to the template. Then we made some changes in order to just pass the necessary keys when we discussed creating the mono repo together with the web developers. That macro `somatic_variant_table(variants` was created after and all keys were there, so it was never tested with unprocessed case and variants directly from the database.